libs/designer/src/lib/core/state/operation/operationSelector.ts

```diff
@@ -22,14 +22,18 @@ export const getNodeOperationData = (state: OperationMetadataState, nodeId: stri
     nodeInputs: getRecordEntry(state.inputParameters, nodeId),
     nodeOutputs: getRecordEntry(state.outputParameters, nodeId),
     nodeDependencies: getRecordEntry(state.dependencies, nodeId),
-    operationMetadata: getRecordEntry(state.operationMetadata, nodeId),
+    operationMetadata: getNodeOperationMetadata(state, nodeId),
     settings: getRecordEntry(state.settings, nodeId),
     staticResult: getRecordEntry(state.staticResults, nodeId),
     actionMetadata: getRecordEntry(state.actionMetadata, nodeId),
     repetitionInfo: getRecordEntry(state.repetitionInfos, nodeId),
   };
 };
 
+export const getNodeOperationMetadata = (state: OperationMetadataState, nodeId: string) => {
+  return getRecordEntry(state.operationMetadata, nodeId);
```

Comment:
is there any reason why this was moved into its own function?

Reply:
For copy scope action, I had to add the nodeData - but only the operation metadata part was needed so didn't want to store the rest of unused data